Framework/DataHandling/src/ISISDataArchive.cpp

```diff
@@ -55,19 +54,22 @@ ISISDataArchive::getArchivePath(const std::set<std::string> &filenames,
     g_log.debug() << ext << ")\n";
   }
 
-  for (const auto &ext : exts) {
-    for (const auto &filename : filenames) {
-      const std::string fullPath = getPath(filename + ext);
+  for (const auto &filename : filenames) {
+    const std::string path_without_extension = getPath(filename);
+    if (!path_without_extension.empty()) {
+      std::string fullPath = getCorrectExtension(path_without_extension, exts);
       if (!fullPath.empty())
         return fullPath;
-    } // it
-  }   // ext
+    }
+  }
   return "";
 }
 
 /**
- * Calls a web service to get a full path to a file.
- * Only returns a full path string if the file exists
+ * Calls a web service to get a path to a file.
+ * Only returns a path string if the file exists.
```

Comment:
I don't think this comment is valid? As we've observed, I think it is guaranteed to return some directory but the file might still not exist there?